### ISSUE TITLE
 Error sending to specific group

### ISSUE THREAD

**Issue by iexos** [AsamK/signal-cli#1252]
## Problem

- my bot is posting once a week into two groups: `test` & `prod`
- sending to `test` works as expected
- sending to `prod` stopped working (exact same command except for different group id)
- `prod` is a busy group with dozens of people, while `test` only exists for the bot messages
- no update of signal-cli since last time it worked
- used signal-cli-native, but also tried with signal-cli-jre (same thing)
- not sure if related/relevant: receiving now sometimes errors out, but can be finished by repeatedly trying. maybe resulted in a broken group state?

Any hints would be appreciated!

## Setup
arch: arm64
deb source: https://packaging.gitlab.io/signal-cli
pkg: signal-cli-native, signal-cli-jre
version: 0.11.10-2

## Logs

Error with signal-cli-jre:
```
Failed to send message: org.signal.libsignal.protocol.InvalidKeyException: invalid signature detected (IOException)
```

Error with signal-cli-native:
```
java.lang.IllegalArgumentException: Invalid object handle
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.JNIObjectHandles.throwIllegalArgumentException(JNIObjectHandles.java:164)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.JNIObjectHandles.getObjectSlow0(JNIObjectHandles.java:159)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.JNIObjectHandles.getObjectSlow(JNIObjectHandles.java:152)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.JNIObjectHandles.getObject(JNIObjectHandles.java:147)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.JNIGeneratedMethodSupport.unboxHandle(JNIGeneratedMethodSupport.java:75)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.functions.JNIFunctions.GetByteArrayElements(JNIFunctions.java:0)                            
        at org.signal.libsignal.internal.Native.PreKeyBundle_New(Native.java)          
        at org.signal.libsignal.protocol.state.PreKeyBundle.<init>(PreKeyBundle.java:36)                    
        at org.whispersystems.signalservice.internal.push.PushServiceSocket.getPreKey(PushServiceSocket.java:732)             
        at org.whispersystems.signalservice.api.SignalServiceMessageSender.handleMismatchedDevices(SignalServiceMessageSender.java:2557)   
        at org.whispersystems.signalservice.api.SignalServiceMessageSender.sendGroupMessage(SignalServiceMessageSender.java:2307)                                        
        at org.whispersystems.signalservice.api.SignalServiceMessageSender.sendGroupDataMessage(SignalServiceMessageSender.java:580)                                      
        at org.asamk.signal.manager.helper.SendHelper.lambda$sendGroupMessage$12(SendHelper.java:356)                                                                     
        at org.asamk.signal.manager.helper.SendHelper.sendGroupMessageInternalWithSenderKey(SendHelper.java:590)                                                          
        at org.asamk.signal.manager.helper.SendHelper.sendGroupMessageInternal(SendHelper.java:452)                                                                       
        at org.asamk.signal.manager.helper.SendHelper.sendGroupMessage(SendHelper.java:392)                                                                               
        at org.asamk.signal.manager.helper.SendHelper.sendAsGroupMessage(SendHelper.java:314)                                                                             
        at org.asamk.signal.manager.helper.SendHelper.sendAsGroupMessage(SendHelper.java:106)                                                                             
        at org.asamk.signal.manager.ManagerImpl.sendMessage(ManagerImpl.java:497)                                                                                         
        at org.asamk.signal.manager.ManagerImpl.sendMessage(ManagerImpl.java:470)                                                                                         
        at org.asamk.signal.manager.ManagerImpl.sendMessage(ManagerImpl.java:591)                                                                                         
        at org.asamk.signal.commands.SendCommand.handleCommand(SendCommand.java:207)                                                                                      
        at org.asamk.signal.App.handleLocalCommand(App.java:289)                                                                                                          
        at org.asamk.signal.App.init(App.java:229)                                                                                                                        
        at org.asamk.signal.Main.main(Main.java:60) 
```

**Comment by AsamK** [comment]
Can you provide the stack trace of the jre version? With `-v`.
It looks like one of the group members has an invalid prekey ...

**Comment by iexos** [comment]
Log of jre with `-v`:
```
2023-05-23T10:11:05.569+0200 [main] INFO  LibSignal - [libsignal]: rust/bridge/jni/src/logging.rs:156: Initializing libsignal version:0.24.0
2023-05-23T10:11:06.592+0200 [main] INFO  com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Starting...
2023-05-23T10:11:07.036+0200 [main] INFO  com.zaxxer.hikari.pool.HikariPool - HikariPool-1 - Added connection org.sqlite.jdbc4.JDBC4Connection@xxxxxxxx
2023-05-23T10:11:07.043+0200 [main] INFO  com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Start completed.
2023-05-23T10:11:10.371+0200 [main] DEBUG LibSignal - [SignalServiceMessageSender]: Using pipe to retrieve attachment upload attributes...
2023-05-23T10:11:10.564+0200 [main] WARN  LibSignal - [SignalServiceMessageSender]: [uploadAttachmentV2] Pipe unavailable, falling back... (WebSocketUnavailableException: WebSocket not currently available.)
2023-05-23T10:11:10.565+0200 [main] DEBUG LibSignal - [SignalServiceMessageSender]: Not using pipe to retrieve attachment upload attributes...
2023-05-23T10:11:13.277+0200 [main] DEBUG o.a.signal.manager.helper.SendHelper - Can use sender key for 55/63 recipients.
2023-05-23T10:11:13.646+0200 [main] DEBUG LibSignal - [SignalServiceMessageSender]: [xxxxxxxxxxxxx] Sending a group data message to 55 recipients using DistributionId xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
2023-05-23T10:11:13.712+0200 [main] INFO  LibSignal - [SignalServiceMessageSender]: Including existing attachment pointer...
2023-05-23T10:11:14.190+0200 [main] INFO  LibSignal - [SignalServiceMessageSender]: [sendGroupMessage][xxxxxxxxxxxxx] Pipe unavailable, falling back... (WebSocketUnavailableException: WebSocket not currently available.)
2023-05-23T10:11:14.978+0200 [main] WARN  LibSignal - [SignalServiceMessageSender]: [sendGroupMessage][xxxxxxxxxxxxx] Handling mismatched devices. (StatusCode: 409)
2023-05-23T10:11:14.988+0200 [main] WARN  LibSignal - [SignalServiceMessageSender]: [handleMismatchedDevices] Address: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx, ExtraDevices: [], MissingDevices: [4]
2023-05-23T10:11:15.216+0200 [main] INFO  com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Shutdown initiated...
2023-05-23T10:11:15.224+0200 [main] INFO  com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Shutdown completed.
2023-05-23T10:11:15.224+0200 [msl-cleanup] DEBUG o.a.s.m.s.s.MessageSendLogStore - Stopping msl cleanup thread
Failed to send message: org.signal.libsignal.protocol.InvalidKeyException: invalid signature detected (IOException)
java.io.IOException: org.signal.libsignal.protocol.InvalidKeyException: invalid signature detected
	at org.whispersystems.signalservice.api.SignalServiceMessageSender.handleMismatchedDevices(SignalServiceMessageSender.java:2567)
	at org.whispersystems.signalservice.api.SignalServiceMessageSender.sendGroupMessage(SignalServiceMessageSender.java:2307)
	at org.whispersystems.signalservice.api.SignalServiceMessageSender.sendGroupDataMessage(SignalServiceMessageSender.java:580)
	at org.asamk.signal.manager.helper.SendHelper.lambda$sendGroupMessage$12(SendHelper.java:356)
	at org.asamk.signal.manager.helper.SendHelper.sendGroupMessageInternalWithSenderKey(SendHelper.java:590)
	at org.asamk.signal.manager.helper.SendHelper.sendGroupMessageInternal(SendHelper.java:452)
	at org.asamk.signal.manager.helper.SendHelper.sendGroupMessage(SendHelper.java:392)
	at org.asamk.signal.manager.helper.SendHelper.sendAsGroupMessage(SendHelper.java:314)
	at org.asamk.signal.manager.helper.SendHelper.sendAsGroupMessage(SendHelper.java:106)
	at org.asamk.signal.manager.ManagerImpl.sendMessage(ManagerImpl.java:497)
	at org.asamk.signal.manager.ManagerImpl.sendMessage(ManagerImpl.java:470)
	at org.asamk.signal.manager.ManagerImpl.sendMessage(ManagerImpl.java:591)
	at org.asamk.signal.commands.SendCommand.handleCommand(SendCommand.java:207)
	at org.asamk.signal.App.handleLocalCommand(App.java:289)
	at org.asamk.signal.App.init(App.java:229)
	at org.asamk.signal.Main.main(Main.java:60)
Caused by: org.signal.libsignal.protocol.InvalidKeyException: invalid signature detected
	at org.signal.libsignal.internal.Native.SessionBuilder_ProcessPreKeyBundle(Native Method)
	at org.signal.libsignal.protocol.SessionBuilder.process(SessionBuilder.java:93)
	at org.whispersystems.signalservice.api.crypto.SignalSessionBuilder.process(SignalSessionBuilder.java:24)
	at org.whispersystems.signalservice.api.SignalServiceMessageSender.handleMismatchedDevices(SignalServiceMessageSender.java:2561)
	... 15 more
```

I tried with removing the bot user from the group and readding it, didn't help

**Comment by AsamK** [comment]
I've pushed a commit handles this error and falls back to another way of sending, though that might then have the same issue...
Can you try this development build? https://github.com/AsamK/signal-cli/actions/runs/5060356772

If that still doesn't work you could try to send a direct message to the address noted in the `[handleMismatchedDevices] Address: xxxxxxxxxxxxxxxxxxxxxxxxxxxxx` line. And also try a `send -e` to reset the session with this recipient.

**Comment by iexos** [comment]
unfortunately i cannot use this build with my arm64. i might try from another machine later

**Comment by iexos** [comment]
It is fixed with 0.11.11, thank you :)
